script.js

```diff
@@ -1,12 +1,421 @@
+  let container = Array.from(containerDiv);
+  let searchBar = document.getElementById(`searchInput`);
+
+  //main function of counting after and before keyup
+  resultParagraph.innerHTML = `${codeCorrection(
+    container.length
+  )} / ${codeCorrection(container.length)}`;
+  searchBar.addEventListener(`keyup`, () => {
+    resultParagraph.innerHTML = `${codeCorrection(
+      container.filter((x) => x.style.display === `initial`).length
+    )} / ${codeCorrection(container.length)}`;
+  });
+}
+
+//for up date result out of searchbar
+function updateResult() {
+  setTimeout(function () {
+    let containerDiv = document.getElementsByClassName(`container`);
+    let container = Array.from(containerDiv);
+    let resultParagraph = document.getElementById(`result`);
+    // console.log(container.filter((x) => x.style.display != `none`));
+    // console.log(container);
+    resultParagraph.innerHTML = `${codeCorrection(
+      container.filter((x) => x.style.cssText === `display: initial;`).length
+    )} / ${codeCorrection(
+      container.filter((x) => x.style.cssText !== `display: none;`).length
+    )}`;
+    // console.log(container.filter((x) => x.style.cssText === `display: initial;`));
+  }, 400);
+}
+
+// select option part
+function selectEpisode(episode) {
+  let select = document.createElement(`select`);
+  header.appendChild(select);
+  select.setAttribute(`id`, `episodeSelect`);
+  select.setAttribute(`class`, `select`);
+  let firstOption = document.createElement(`option`);
+  select.appendChild(firstOption);
+  firstOption.setAttribute(`id`, `firstEpisode`);
+  firstOption.setAttribute(`value`, `select an option`);
+  firstOption.textContent = `Episode Selector:`;
+  episode.forEach((x) => {
+    let option = document.createElement(`option`);
+    option.setAttribute(`value`, x.id);
+    option.setAttribute(`class`, `option`);
+    select.appendChild(option);
+    option.textContent = `S${codeCorrection(x.season)}E${codeCorrection(
+      x.number
+    )} - ${x.name}`;
+  });
+
+  select.addEventListener(`change`, (e) => {
+    let targetDiv = document.getElementById(`${e.target.value}`);
+    let containerDiv = document.getElementsByClassName("container");
+    container = Array.from(containerDiv);
+    container.map((x) => (x.style.display = `none`));
+    targetDiv.style.display = `initial`;
+    updateResult();
+  });
+}
+
+function selectShow(show) {
+  let select = document.createElement(`select`);
+  header.appendChild(select);
+  select.setAttribute(`id`, `showSelect`);
+  select.setAttribute(`class`, `select`);
+  let firstOption = document.createElement(`option`);
+  select.appendChild(firstOption);
+  firstOption.setAttribute(`id`, `firstShow`);
+  firstOption.setAttribute(`value`, `select an option`);
+  firstOption.textContent = `Show Selector:`;
+
+  //sorting  in alphabetical order case-insensitive
+  let showArray = Array.from(show);
+  let sort = showArray.sort(function (a, b) {
```

Comment:
Nice sort function, but i think a better variable name would be maybe something like `sortedItems` instead of `sort`, so it's more clear the contents of this variable